blueflood-core/src/test/java/com/rackspacecloud/blueflood/service/BluefloodServiceStarterTest.java

```diff
@@ -0,0 +1,275 @@
+        config.setProperty(CoreConfig.QUERY_MODE, "true");
+        config.setProperty(CoreConfig.QUERY_MODULES, "");
+        String[] args = new String[0];
+
+        // when
+        BluefloodServiceStarterException ex = null;
+        try {
+            BluefloodServiceStarter.main(args);
+        } catch (BluefloodServiceStarterException e) {
+            ex = e;
+        }
+
+        // then
+        assertNotNull(ex);
+        assertEquals(1, ex.getStatus());
+    }
+
+    @Test
+    public void testQueryModeEnabledWithDummyModule() {
+
+        // given
+        Configuration config = Configuration.getInstance();
+        config.setProperty(CoreConfig.QUERY_MODE, "true");
+        config.setProperty(CoreConfig.QUERY_MODULES, "com.rackspacecloud.blueflood.service.DummyQueryService");
+        String[] args = new String[0];
+
+        // when
+        BluefloodServiceStarterException ex = null;
+        try {
+            BluefloodServiceStarter.main(args);
+        } catch (BluefloodServiceStarterException e) {
+            ex = e;
+        }
+
+        // then
+        assertNull(ex);
+
+        assertNotNull(DummyQueryService.getInstances());
+        assertEquals(1, DummyQueryService.getInstances().size());
+        assertNotNull(DummyQueryService.getMostRecentInstance());
+
+        assertTrue(DummyQueryService.getMostRecentInstance().getStartServiceCalled());
+    }
+
+    @Test
+    public void testRollupModeEnabled() {
+
+        // given
+        Configuration config = Configuration.getInstance();
+        config.setProperty(CoreConfig.ROLLUP_MODE, "true");
+        String[] args = new String[0];
+
+        // when
+        BluefloodServiceStarterException ex = null;
+        try {
+            BluefloodServiceStarter.main(args);
+        } catch (BluefloodServiceStarterException e) {
+            ex = e;
+        }
+
+        // then
+        assertNull(ex);
+    }
+
+    @Test
+    public void testEventListenerServiceEnabledWithDummyModule() {
+
+        // given
+        Configuration config = Configuration.getInstance();
+        config.setProperty(CoreConfig.EVENT_LISTENER_MODULES, "com.rackspacecloud.blueflood.service.DummyEventListenerService");
+        String[] args = new String[0];
+
+        // when
+        BluefloodServiceStarterException ex = null;
+        try {
+            BluefloodServiceStarter.main(args);
+        } catch (BluefloodServiceStarterException e) {
+            ex = e;
+        }
+
+        // then
+        assertNull(ex);
+
+        assertNotNull(DummyEventListenerService.getInstances());
+        assertEquals(1, DummyEventListenerService.getInstances().size());
+        assertNotNull(DummyEventListenerService.getMostRecentInstance());
+
+        assertTrue(DummyEventListenerService.getMostRecentInstance().getStartServiceCalled());
+    }
+
+    @After
+    public void tearDown() throws IOException {
+        Configuration config = Configuration.getInstance();
+        config.init();
```

Comment:
To reset the configuration for other tests. Different tests were interacting with the configuration and not cleaning up after themselves, which interfered with other tests. By calling init both before and after, we make sure that the tests are isolated.